pkg/controller/datadogagentdeployment/agent.go

```diff
@@ -352,3 +350,34 @@ func newDaemonsetObjectMetaData(agentdeployment *datadoghqv1alpha1.DatadogAgentD
 		Annotations: annotations,
 	}
 }
+
+func getAgentCustomConfigConfigMapName(dad *datadoghqv1alpha1.DatadogAgentDeployment) string {
+	return fmt.Sprintf("%s-datadog-yaml", dad.Name)
+}
+
+func buildAgentConfigurationConfigMap(dad *datadoghqv1alpha1.DatadogAgentDeployment) (*corev1.ConfigMap, error) {
+	if dad.Spec.Agent.CustomConfig == nil {
+		return nil, nil
+	}
+
+	// Validate that user input is valid YAML
+	// Maybe later we can implement that directly verifies against Agent configuration?
+	m := make(map[interface{}]interface{})
+	if err := yaml.Unmarshal([]byte(*dad.Spec.Agent.CustomConfig), m); err != nil {
+		return nil, err
```

Comment:
I think it can be good to wrap this `yaml.Unmarshal` err with a bit more context, like it is related to the CustomConfig file.

Reply:
Yep, that makes sense, will do